### PR TITLE
Gradle dependencies cleanup

### PR DESCRIPTION
**Pull request by iNikem**
Partially adresses #351 .

This PR touches mostly "shared" modules, like `auto-bootstrap` or `java.gradle` to allow for meaningful review. Later PR will change "leave" modules, where changes are very mechanical.

### CODE REVIEW

**Comment by iNikem** [comment]
But I don't understand why muzzle fails :(

**Comment by trask** [comment]
I think `muzzle` found a real issue. Not really sure why it didn't catch it before, probably your dependency cleanup revealed it somehow.

To make `muzzle` pass for now, go ahead and lie to it:

<pre>
muzzle {
  pass {
    group = "io.opentracing.contrib.dropwizard"
    module = "dropwizard-opentracing"
    versions = "(,)"
    <b>extraDependency 'io.opentelemetry:opentelemetry-contrib-auto-annotations:0.5.0'</b>
  }
}
</pre>

i think the "correct" way to deal with this is to split out the `WithSpan` instrumentation into a separate module, e.g. https://github.com/trask/opentelemetry-java-instrumentation/commit/a8e7e8227689e0bff7beb11735c9d6d171e97716

i'll turn this prototype into a follow-on PR after this one is merged

**Comment by iNikem** [comment]
@trask can you explain why current approach does not work? What is wrong with it?

**Comment by trask** [comment]
`WithSpan` should be treated like a third-party library that we are instrumenting, e.g.

https://github.com/trask/opentelemetry-java-instrumentation/blob/split-annotation-module/instrumentation/annotations/annotations-opentelemetry/annotations-opentelemetry.gradle#L4-L21

but instead we have a weird mix of treating it as a class we are instrumenting, but also as an internal class, e.g. we add it to the bootstrap classes in `SpockRunner.java` which is incorrect, and we relocate it in `java-agent.gradle` which is also incorrect.

**Comment by iNikem** [comment]
I see... Ok, I mostly understand, but why we have to split annotations instrumentation module? If we should treat `WithSpan` as any other third party library, why should be it separate from other annotations support?

**Comment by trask** [comment]
Oh, I think this is the way to go:

```
muzzle {
  pass {
    group = "io.opentelemetry"
    module = "opentelemetry-contrib-auto-annotations"
    versions = "(,)"
    extraDependency 'io.opentracing.contrib.dropwizard:dropwizard-opentracing:0.2.2'
  }
}
```

I was going to say it's better when there's only a single compile-time dependency in each instrumentation module (and that's the dependency we put in the muzzle block).

But then I realized that the `trace-annotation` module already only has one compile-time dependency, since `WithSpan` is the only annotation we take a compile-time dependency on.

We still need the `extraDependency`, because otherwise muzzle will say that `TraceAnnotationsInstrumentation` failed to match anything on the classpath.

File: instrumentation/trace-annotation/trace-annotation.gradle [hunk 1]
```diff
@@ -5,10 +5,13 @@ muzzle {
     group = "io.opentracing.contrib.dropwizard"
     module = "dropwizard-opentracing"
     versions = "(,)"
+    extraDependency 'io.opentelemetry:opentelemetry-contrib-auto-annotations:0.5.0'
```
Comment (trask) [inline instrumentation/trace-annotation/trace-annotation.gradle:8]:
```suggestion
    group = "io.opentelemetry"
    module = "opentelemetry-contrib-auto-annotations"
    versions = "(,)"
    extraDependency 'io.opentracing.contrib.dropwizard:dropwizard-opentracing:0.2.2'
```

**Pull request merged**